src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1254,7 +1254,7 @@ private static MethodHandle generate(Lookup lookup, MethodType args, String[] co
                 MethodHandlePair handlePair = weakConstructorHandle.get();
                 if (handlePair != null) {
                     try {
-                        var instance = handlePair.constructor.invoke(constants);
+                        var instance = handlePair.constructor.invokeBasic(constants);
```

Comment:
Hmm, `invokeBasic` notes ask to call with plain `Object` but `constants` is a `Object[]`... Don't know how that really worked out.

Reply:
I tried with/without `(Object)constants` and didn't see any difference in instrumentation. It makes it more obviously correct so I'll add it.